modules/packages/ExternalArray.chpl

```diff
@@ -25,6 +25,19 @@
 module ExternalArray {
   use ChapelStandard;
 
+  extern type chpl_free_func;
+  private extern const CHPL_FREE_FUNC_NIL: chpl_free_func;
+
+  extern record chpl_external_array {
+    var elts: c_void_ptr;
+    var size: uint;
+
+    var freer: chpl_free_func;
```

Comment:
`freer` isn't used in the Chapel code.  If we abstracted it to be just a `c_void_ptr` then we could get rid of both the `chpl_free_func` and `CHPL_FREE_FUNC_NIL` externs, couldn't we?  (And either pass `nil` or just elide the third argument to `new chpl_external_array()` in `makeArrayFromPtr()`.)

Reply:
Replying to my own comment: We would then have to cast it when calling via the pointer, but that seems a small price to pay for the added interface information-hiding.